Realm/Realm.Weaver/Analytics/AnalyticsUtils.cs

```diff
@@ -0,0 +1,164 @@
+                            s.ContainsIgnoreCase("macos") ||
+                            s.ContainsIgnoreCase("maccatalyst"):
+                        return Metric.OperatingSystem.MacOS;
+                    case string s when s.ContainsIgnoreCase("tvos"):
+                        return Metric.OperatingSystem.TvOs;
+                    default:
+                        return Metric.OperatingSystem.Windows;
+                }
+            });
+
+        public static string SHA256Hash(byte[] bytes)
+        {
+            using (var sha256 = System.Security.Cryptography.SHA256.Create())
+            {
+                return BitConverter.ToString(sha256.ComputeHash(bytes));
+            }
+        }
+
+        public static void ComputeHostOSNameAndVersion(out string osType, out string osVersion)
+        {
+            var platformRegex = new Regex("^(?<platform>[^0-9]*) (?<version>[^ ]*)", RegexOptions.Compiled);
+            var osDescription = platformRegex.Match(RuntimeInformation.OSDescription);
+            if (osDescription.Success)
+            {
+                osType = osDescription.Groups["platform"].Value;
+                osVersion = osDescription.Groups["version"].Value;
+            }
+            else
+            {
+                osType = Environment.OSVersion.Platform.ToString();
+                osVersion = Environment.OSVersion.VersionString;
+            }
+
+            osType = ConvertOsNameToMetricsVersion(osType);
+        }
+
+        public static string GetHostCpuArchitecture =>
+            WrapInTryCatch(() => ConvertArchitectureToMetricsVersion(RuntimeInformation.ProcessArchitecture.ToString()));
+
+        // TODO andrea: module.Architecture reports "I386" which isn't a value I could find in the documentation of MS. Investigate
+        public static string GetTargetCpuArchitecture(ModuleDefinition module) =>
+            WrapInTryCatch(() => ConvertArchitectureToMetricsVersion(module.Architecture.ToString()));
+
+        public static byte[] GenerateComputerIdentifier =>
+            NetworkInterface.GetAllNetworkInterfaces()
+                .Where(n => n.Name == "en0" || (n.OperationalStatus == OperationalStatus.Up && n.NetworkInterfaceType != NetworkInterfaceType.Loopback))
+                .Select(n => n.GetPhysicalAddress().GetAddressBytes())
+                .FirstOrDefault();
+
+        private static string ConvertOsNameToMetricsVersion(string osName) =>
+            WrapInTryCatch(() =>
+            {
+                switch (osName)
+                {
+                    case string s when s.ContainsIgnoreCase(Metric.OperatingSystem.Windows):
+                        return Metric.OperatingSystem.Windows;
+                    case string s when s.ContainsIgnoreCase(Metric.OperatingSystem.MacOS):
+                        return Metric.OperatingSystem.MacOS;
+                    case string s when s.ContainsIgnoreCase(Metric.OperatingSystem.Linux):
```

Comment:
Does this work for various Linux distro? (Just curious)

Reply:
This is never been tested outside of my windows machine. It could be a good idea to add unit tests to test these details. I'll think about what can be done.

Reply:
Testing for this is going to be part of Phase 2 of this endevour.